client/rest-high-level/src/test/java/org/elasticsearch/client/ml/inference/trainedmodel/tree/TreeNodeTests.java

```diff
@@ -63,6 +64,7 @@ public static TreeNode createRandomLeafNode(double internalValue) {
             .setDefaultLeft(randomBoolean() ? null : randomBoolean())
             .setLeftChild(left)
             .setRightChild(right)
+            .setNumberSamples(randomNonNegativeLong())
```

Comment:
nit: Cover the null case in the test. If you made `public Builder setNumberSamples() take a `Long` then this code could be `.setNumberSamples(randomBoolean() ? null : randomNonNegativeLong())`